### PR TITLE
Export named validations to openApi properties

### LINKED ISSUE #897: Export validation annotations in openapi schema
**Describe the problem/challenge you have**
Currently the openapi schema generated using `ytt -f schema.yaml --data-values-schema-inspect --output openapi-v3` is missing any of the `#@schema/validation` rules.

**Describe the solution you'd like**
Exporting the openapi-v3 schema, should include the [named rules](https://carvel.dev/ytt/docs/v0.48.0/lang-ref-ytt-schema/#named-validation-rules) since they all seem to be relatively easy to map to openapi-v3.

| Type    | ytt validation      | openapi validation | 
| ------- | ------------------- | ------------------ |
| string  | min_len             | minLength          |
| string  | max_len             | maxLength          |
| integer | min                 | minimum            |
| integer | max.                | maximum            |
| any     | not_null=(nullable) | nullable=true      |
| any     | one_of              | enum               |

Only `one_not_null` seems to be a bit more complicate to map, and could be left out.

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by peschmae**
This pull requests, exposes the attributes for the named `#@schema/validations` to the openApi schema, where a direct mapping is possible.

| Type    | ytt validation      | openapi validation | 
| ------- | ------------------- | ------------------ |
| string  | min_len             | minLength          |
| string  | max_len             | maxLength          |
| integer | min                 | minimum            |
| integer | max                 | maximum            |
| any     | one_of              | enum               |

The PR includes tests for all those validations.

Fixes #897

### CODE REVIEW

**Comment by prembhaskal** [comment]
will try to review this week.

**Comment by peschmae** [comment]
Just saw the linting errors, and will look into them.

Should I squash my commits and force push, or is it fine to have mutliple commits, and the will be squashed when merged?

**Comment by prembhaskal** [comment]
you can have multiple commits and then squash them before merging.

**Comment by peschmae** [comment]
Any change to get a review on this?
Compared to #901 this is a smaller change and would allow to integrate the conversion from openapi to json schema, in a separate tool if the other PR wouldn't be merged.

File: pkg/validations/validate.go [hunk 1]
```diff
@@ -56,6 +57,55 @@ type validationKwargs struct {
 	oneOf      starlark.Sequence
 }
 
+// ValidationMap returns a map of the validationKwargs and their values.
+func (v NodeValidation) ValidationMap() map[string]interface{} {
+
```
Comment (joaopapereira) [inline pkg/validations/validate.go]: nit:
```suggestion
```

File: pkg/validations/validate.go [hunk 2]
```diff
@@ -56,6 +57,55 @@ type validationKwargs struct {
...
+			default:
+				val = x.String()
+			}
+
+			enum = append(enum, val)
+
+		}
+
+		validations["oneOf"] = enum
+
```
Comment (joaopapereira) [inline pkg/validations/validate.go]: nit:
```suggestion
```

File: pkg/validations/validate.go [hunk 3]
```diff
@@ -56,6 +57,55 @@ type validationKwargs struct {
...
+	}
+
+	return validations
+
```
Comment (joaopapereira) [inline pkg/validations/validate.go]: nit:
```suggestion
```

File: pkg/validations/validate.go [hunk 4]
```diff
@@ -56,6 +57,55 @@ type validationKwargs struct {
...
+			case "string":
+				val, _ = strconv.Unquote(x.String())
+			case "int":
+				val, _ = strconv.Atoi(x.String())
+			default:
+				val = x.String()
+			}
+
+			enum = append(enum, val)
+
```
Comment (joaopapereira) [inline pkg/validations/validate.go]: nit: 
```suggestion
```

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Added some nitpick changes more stylistic than anything.
Generally, the PR looks good.

**Comment by peschmae** [comment]
Ammended whitespace cleanup into latest commit, and force pushed

**Review by sethiyash — APPROVED** [review]
LGTM!

**Review by cppforlife — APPROVED** [review]

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**